htlcswitch/payment_result_test.go

```diff
@@ -193,4 +192,26 @@ func TestNetworkResultStore(t *testing.T) {
 			t.Fatalf("no result received")
 		}
 	}
+
+	// Clean the store keeping the first two results.
+	toKeep := map[uint64]struct{}{
+		0: struct{}{},
+		1: struct{}{},
+	}
+	// Finally, delete the result.
+	if err := store.cleanStore(toKeep); err != nil {
+		t.Fatal(err)
+	}
+
+	// Payment IDs 0 and 1 should be found, 2 and 3 should be deleted.
+	for i := uint64(0); i < numResults; i++ {
+		_, err = store.getResult(i)
+		if i <= 1 && err != nil {
+			t.Fatalf("unable to get result: %v", err)
+		}
+		if i >= 2 && err != ErrPaymentIDNotFound {
+			t.Fatalf("expected ErrPaymentIDNotFound, got %v", err)
```

Comment:
Similarly, you could use `require.Error`.

Reply:
I don't know if that can be used to match on the exact error?

Reply:
this should do it: `require.Error(t, ErrPaymentIDNotFound, err)`

Reply:
Couldn't get it to work. Looks like `require.Error` only checks that it is non-nil.